The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11920653831/job/33223085383) [(plain)](https://github.com/rust-lang/rust/commit/d98b13910d981c7fcff6263eca9fc59f121a53c8/checks/33223085383/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.880 Building wheels for collected packages: reuse
#13 2.881   Building wheel for reuse (pyproject.toml): started
#13 3.138   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.139   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.139   Stored in directory: /tmp/pip-ephem-wheel-cache-5rur4sy8/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.142 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.567 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.567 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.124 Collecting virtualenv
#13 4.124 Collecting virtualenv
#13 4.163   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 4.226      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 51.6 MB/s eta 0:00:00
#13 4.267 Collecting distlib<1,>=0.3.7
#13 4.270   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.279      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 73.5 MB/s eta 0:00:00
#13 4.314 Collecting platformdirs<5,>=3.9.1
#13 4.326   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.436 Collecting filelock<4,>=3.12.2
#13 4.441   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.527 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.711 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      317376 kB
DirectMap2M:     8071168 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 32.30s
##[endgroup]
fmt check
Diff in /checkout/compiler/rustc_codegen_ssa/src/back/link.rs:1108:
         match (strip, crate_type) {
             (Strip::Debuginfo, _) => {
                 let strip_options = vec!["-S".to_string()];
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some(&strip_options))
+                    sess,
+                    stripcmd,
+                    out_filename,
+                    Some(&strip_options),
+                    Some(&strip_options),
+                )
             }
             // Per the manpage, `-x` is the maximum safe strip level for dynamic libraries. (#93988)
             (Strip::Symbols, CrateType::Dylib | CrateType::Cdylib | CrateType::ProcMacro) => {
Diff in /checkout/compiler/rustc_codegen_ssa/src/back/link.rs:1115:
                 let strip_options = vec!["-x".to_string()];
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some(&strip_options))
+                    sess,
+                    stripcmd,
+                    out_filename,
+                    Some(&strip_options),
+                    Some(&strip_options),
+                )
             }
             (Strip::Symbols, _) => {
                 strip_symbols_with_external_utility(sess, stripcmd, out_filename, None)
Diff in /checkout/compiler/rustc_codegen_ssa/src/back/link.rs:1134:
             // Always preserve the symbol table (-x).
             Strip::Debuginfo => {
                 let strip_options = vec!["-x".to_string()];
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some(&strip_options))
+                    sess,
+                    stripcmd,
+                    out_filename,
+                    Some(&strip_options),
+                    Some(&strip_options),
+                )
             }
             // Strip::Symbols is handled via the --strip-all linker option.
             Strip::Symbols => {}
Diff in /checkout/compiler/rustc_codegen_ssa/src/back/link.rs:1153:
             Strip::Debuginfo => {
                 // FIXME: AIX's strip utility only offers option to strip line number information.
                 strip_options.push("-l".to_string());
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some(&strip_options))
+                    sess,
+                    stripcmd,
+                    out_filename,
+                    Some(&strip_options),
+                    Some(&strip_options),
+                )
             }
             Strip::Symbols => {
                 // Must be noted this option might remove symbol __aix_rust_metadata and thus removes .info section which contains metadata.
Diff in /checkout/compiler/rustc_codegen_ssa/src/back/link.rs:1160:
                 strip_options.push("-r".to_string());
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some(&strip_options))
+                    sess,
+                    stripcmd,
+                    out_filename,
+                    Some(&strip_options),
+                    Some(&strip_options),
+                )
             }
             Strip::None => {}
         }
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_query_system/src/query/caches.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/linker.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/symbol_export.rs" "/checkout/compiler/rustc_query_system/src/query/job.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/write.rs" "/checkout/compiler/rustc_query_system/src/query/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/rpath.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/lto.rs" "/checkout/compiler/rustc_query_system/src/error.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/rpath/tests.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/apple/tests.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/metadata.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/archive.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/command.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/link.rs" "/checkout/compiler/rustc_codegen_ssa/src/lib.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/dep_node.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/edges.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/serialized.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/debug.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/query.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/graph.rs" "/checkout/compiler/rustc_query_system/src/dep_graph/mod.rs" "/checkout/compiler/rustc_query_system/src/lib.rs" "/checkout/compiler/rustc_infer/src/traits/engine.rs" "/checkout/compiler/rustc_infer/src/traits/structural_impls.rs" "/checkout/compiler/rustc_infer/src/traits/mod.rs" "/checkout/compiler/rustc_infer/src/traits/project.rs" "/checkout/compiler/rustc_infer/src/traits/util.rs" "/checkout/compiler/rustc_infer/src/infer/free_regions.rs" "/checkout/compiler/rustc_infer/src/infer/projection.rs" "/checkout/compiler/rustc_infer/src/infer/region_constraints/leak_check.rs" "/checkout/compiler/rustc_infer/src/infer/region_constraints/mod.rs" "/checkout/compiler/rustc_hir_typeck/src/method/probe.rs" "/checkout/compiler/rustc_hir_typeck/src/method/confirm.rs" "/checkout/compiler/rustc_hir_typeck/src/method/suggest.rs" "/checkout/compiler/rustc_hir_typeck/src/method/prelude_edition_lints.rs" "/checkout/compiler/rustc_hir_typeck/src/method/mod.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/canonicalizer.rs" "/checkout/compiler/rustc_hir_typeck/src/closure.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/query_response.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/instantiate.rs" "/checkout/compiler/rustc_infer/src/infer/canonical/mod.rs" "/checkout/compiler/rustc_infer/src/infer/type_variable.rs" "/checkout/compiler/rustc_infer/src/infer/opaque_types/table.rs" "/checkout/compiler/rustc_infer/src/infer/opaque_types/mod.rs" "/checkout/compiler/rustc_infer/src/infer/at.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/arg_matrix.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/inspect_obligations.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs" "/checkout/compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs" "/checkout/compiler/rustc_hir_typeck/src/_match.rs" "/checkout/compiler/rustc_hir_typeck/src/gather_locals.rs" "/checkout/compiler/rustc_hir_typeck/src/pat.rs" "/checkout/compiler/rustc_hir_typeck/src/intrinsicck.rs" "/checkout/compiler/rustc_hir_typeck/src/fallback.rs" "/checkout/compiler/rustc_hir_typeck/src/expectation.rs" "/checkout/compiler/rustc_hir_typeck/src/rvalue_scopes.rs" "/checkout/compiler/rustc_hir_typeck/src/autoderef.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/apple.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Nov 19 20:03:30 UTC 2024
  network time: Tue, 19 Nov 2024 20:03:30 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_codegen_ssa/src/back/link.rs
+++ b/compiler/rustc_codegen_ssa/src/back/link.rs
@@ -1107,14 +1107,14 @@ fn link_natively(
         let stripcmd = "rust-objcopy";
         match (strip, crate_type) {
             (Strip::Debuginfo, _) => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some("-S"))
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-S"])
             }
             // Per the manpage, `-x` is the maximum safe strip level for dynamic libraries. (#93988)
             (Strip::Symbols, CrateType::Dylib | CrateType::Cdylib | CrateType::ProcMacro) => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some("-x"))
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-x"])
             }
             (Strip::Symbols, _) => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, None)
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &[])
             }
             (Strip::None, _) => {}
         }
@@ -1131,7 +1131,7 @@ fn link_natively(
         match strip {
             // Always preserve the symbol table (-x).
             Strip::Debuginfo => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some("-x"))
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-x"])
             }
             // Strip::Symbols is handled via the --strip-all linker option.
             Strip::Symbols => {}
@@ -1148,11 +1148,15 @@ fn link_natively(
         match strip {
             Strip::Debuginfo => {
                 // FIXME: AIX's strip utility only offers option to strip line number information.
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some("-l"))
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &[
+                    "-X32_64", "-l",
+                ])
             }
             Strip::Symbols => {
                 // Must be noted this option might remove symbol __aix_rust_metadata and thus removes .info section which contains metadata.
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, Some("-r"))
+                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &[
+                    "-X32_64", "-r",
+                ])
             }
             Strip::None => {}
         }
@@ -1165,12 +1169,10 @@ fn strip_symbols_with_external_utility(
     sess: &Session,
     util: &str,
     out_filename: &Path,
-    option: Option<&str>,
+    options: &[&str],
 ) {
     let mut cmd = Command::new(util);
-    if let Some(option) = option {
-        cmd.arg(option);
-    }
+    cmd.args(options);
 
     let mut new_path = sess.get_tools_search_paths(false);
     if let Some(path) = env::var_os("PATH") {

